src/compressed_tensors/quantization/lifecycle/apply.py

```diff
@@ -71,13 +74,20 @@ def apply_quantization_status(model: Module, status: QuantizationStatus):
     :param model: model to apply quantization to
     :param status: status to update the module to
     """
-    if status >= QuantizationStatus.INITIALIZED:
+    current_status = _infer_status(model)
+
+    if status >= QuantizationStatus.INITIALIZED > current_status:
         model.apply(initialize_module_for_quantization)
-    if status >= QuantizationStatus.CALIBRATION:
+
+    if current_status < status >= QuantizationStatus.CALIBRATION > current_status:
```

Comment:
I'm not used to if statements formatted this way, how does it resolve? Is this the same as:

`(current_status < status) and (status > QuantizationStatus.CALIBRATION) and (QuantizationStatus.CALIBRATION > current_status)`?

Reply:
this is a mistake - first check against current_status should be deleted